emb/network/transport/transporter.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <concepts>
+#include <inttypes.h>
+#include <span>
+
+namespace emb {
+namespace network {
+namespace transport {
+
+// Concept for a transport layer
+template <typename T>
+concept TransporterLike = requires(T t, std::span<uint8_t> data) {
```

Comment:
concepts are cool!

Reply:
Super confusing tbh, but better than nothing. I tried making a `ProjectLike` syntax but quickly got into a rabbit hole trying to forward reference the `Node` class, which is also templated by said `ProjectLike` entries. I found some great article about "C++ concepts are not [x]" (forgot what `[x]` was) that basically boiled down to "you can't use Python's equivalent of `Any` to say 'anything can go here, just make sure it takes _something_'". That's separate from what I did in `serializer.hpp` where the concept says "take this dummy struct and make sure it's a valid input into this templated function"

`Any`-ways, I got good mileage out of it where I could.